### ISSUE TITLE
Are updates to Order Book provided in real-time or using a polling interval?

### ISSUE THREAD

**Issue by mellertson** [slazarov/python-bittrex-websocket#75]
First of all, thanks so much for sharing your work.  I've been looking at your code in both the `bittrex-webocket` and `bittrex-websocket-aio` repos.  You've done brilliant work on both!

I'm particularly interested in using the order book syncing available in your `bittrex-websocket` library.  But, I need to make sure whatever library I use provides truly real-time updates to the order book.  I made a small modification (below) to the example posted in in the docs.  My goal was to determine if updates are streamed in real-time as they are changed in Bittrex's order book.  I noticed that it seems to print a message out approximately every 1 second when `on_public()` is called.  I also noticed sometimes it seems to print out a message a bit less than a second, but I am not 100% sure, so I figured I'd ask.

Does your `bittrex-websocket` library provide truly real-time updates to the order book? Or is the order book updated on a polling interval?

```python
from __future__ import print_function
from time import sleep
from bittrex_websocket import OrderBook

def main():
	class MySocket(OrderBook):
		def on_public(self, msg):  # here's my modified couple lines :-)
			super(MySocket, self).on_public(msg=msg)
			print('Received order book update for {}'.format(msg))

	# Create the socket instance
	ws = MySocket()
	# Enable logging
	ws.enable_log()
	# Define tickers
	tickers = ['BTC-ETH']
	# Subscribe to order book updates
	ws.subscribe_to_orderbook(tickers)

	while True:
		sleep(9999)
		book = ws.get_order_book('BTC-ETH')
		print(f"Printing order book for BTC-ETH: {book[u'S'][0]}")
	else:
		pass


if __name__ == "__main__":
	main()
```

**Comment by mellertson** [comment]
I swear, every time I ask a question Murphy's law kicks in and WAM, I find the answer.  lol

After I posted my question I was parusing through Bittrex's docs, and I believe I found my answer.  Here it is for anyone else who might have the same question.

> [**Tracking an Order Book**](https://github.com/Bittrex/bittrex.github.io#tracking-an-order-book)
Many users poll the REST API as quickly as possible for updated order data. At very high call rates, the API most often returns the same data because we only update it **once per second**. By using the WS API’s QueryExchangeState and SubscribeToExchangeDeltas functions, developers can maintain their own copies of one or more order books.
